### PR TITLE
dlm/dont-end-session-if-user-hits-n

### PR DESCRIPTION
**Pull request by dleviminzi**
When using `beam serve` if you try to kill the session and select "n" when prompted you end up with orphan serves. With this change, selecting "n" leaves the terminal session alive. This allows users to continue to see file syncing and updates. It also means that when the user actually decides to exit later on, the serve will be ended.

### CODE REVIEW

**Review by luke-lombardi — CHANGES_REQUESTED** [review]

File: sdk/src/beta9/abstractions/endpoint.py [hunk 1]
```diff
@@ -208,6 +209,9 @@ def local(self, *args, **kwargs) -> Any:
 
     @with_grpc_error_handling
     def serve(self, timeout: int = 0):
+        signal.signal(signal.SIGINT, sigint_handler)
+        threading.current_thread().shutdown_flag = False
```
Comment (jsun-m) [inline sdk/src/beta9/abstractions/endpoint.py:213]: This shutdown flag is a custom variable? I don't seem to see it in the threading class

File: sdk/src/beta9/abstractions/endpoint.py [hunk 2]
```diff
@@ -218,40 +222,38 @@ def serve(self, timeout: int = 0):
         ):
             return False
 
-        try:
-            with terminal.progress("Serving endpoint..."):
```
[the remark this answers is not in the document]
Comment (dleviminzi) [reply sdk/src/beta9/abstractions/endpoint.py:222]: The console status does not play nicely with the intercept handler and prompt. I tried a few different ways to "pause" and "resume" it. I couldn't find anyway that worked.

File: sdk/src/beta9/abstractions/endpoint.py (hunk 1 above)
Comment (dleviminzi) [reply sdk/src/beta9/abstractions/endpoint.py:213]: Yeah we're using this to flag shutdowns. The default sigint handling kills the gRPC connection and we don't want that to happen until user intent is confirmed.

File: sdk/src/beta9/abstractions/endpoint.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dleviminzi) [reply sdk/src/beta9/abstractions/endpoint.py:222]: If the status is inside the `while serve_thread.is_alive():` loop it flickers bc its being restarted all the time. If its above the loop, the termination prompt is immediately overwritten and user input is never collected.

[the remark this answers is not in the document]
Comment (luke-lombardi) [reply sdk/src/beta9/abstractions/endpoint.py:222]: I don't really want to merge this if we can't get it working with the terminal progress. @mernit thoughts?

**Pull request closed without merging**